src/js/jsx/shared/ColorInput.jsx

```diff
@@ -52,6 +53,60 @@ define(function (require, exports, module) {
         { key: os.eventKeyCode.TAB, modifiers: null }
     ];
 
+    var ColorSwatch = React.createClass({
+        mixins: [FluxMixin, Focusable],
+
+        /**
+         * On escape, release focus and blur the swatch.
+         *
+         * @private
+         * @param {SyntheticEvent} event
+         */
+        _handleKeyDown: function (event) {
+            var key = event.key;
+
+            switch (key) {
+                case "Escape":
+                    this.releaseFocus()
+                        .bind(this)
+                        .then(function () {
+                            React.findDOMNode(this).blur();
+                        });
+                    break;
+            }
+
+            this.props.onKeyDown(event);
```

Comment:
part of me thinks this should validate the existence of this prop before calling it, since it is not otherwise required by react.  But I realize this is a "private" component, so I'll leave it up to you.